src/backend/distributed/sql/udfs/citus_finalize_upgrade_to_citus11/latest.sql

```diff
@@ -0,0 +1,184 @@
+-- citus_finalize_upgrade_to_citus11() is a helper UDF ensures
+-- the upgrade to Citus 11 is finished successfully. Upgrade to
+-- Citus 11 requires all active primary worker nodes to get the
+-- metadata. And, this function's job is to sync the metadata to
+-- the nodes that does not already have
+-- once the function finishes without any errors and returns true
+-- the cluster is ready for running distributed queries from
+-- the worker nodes. When debug is enabled, the function provides
+-- more information to the user.
+CREATE OR REPLACE FUNCTION pg_catalog.citus_finalize_upgrade_to_citus11()
```

Comment:
I'm wondering whether we should have a generic post-upgrade function in case there is something we want to change in the future as well (e.g. fix metadata inconsistencies), but not leaning in any particular direction.

Reply:
I guess for those kinds of scenarios, it might make sense to have `verify_metadata` UDF (and useful in general). Which is maybe reports the differences. Plus, another `sync_missing_metadata` which is smart enough to sync only the missing metadata. 

The reason I refrain to add such a generic post-upgrade UDF is the feeling of diverging from a regular extension to something a lot more complicated :) I know we have such thing for PG upgrades, but that sounds more acceptable than this.

But, similar to you, I don't have strong opinions on any direction and can change if you suggest so.